x-pack/plugins/triggers_actions_ui/public/application/lib/rule_api/load_execution_log_aggregations.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import { HttpSetup } from 'kibana/public';
+import { INTERNAL_BASE_ALERTING_API_PATH } from '../../constants';
+import { IExecutionLogResult, IExecutionLog } from '../../../../../alerting/common';
+import { AsApiContract, RewriteRequestCase } from '../../../../../actions/common';
+
+const getRenamedLog = (data: IExecutionLog) => {
+  const {
+    duration_ms,
+    total_search_duration_ms,
+    es_search_duration_ms,
+    schedule_delay_ms,
+    ...rest
+  } = data;
+
+  return {
+    execution_duration: data.duration_ms,
+    total_search_duration: data.total_search_duration_ms,
+    es_search_duration: data.es_search_duration_ms,
+    schedule_delay: data.schedule_delay_ms,
+    ...rest,
+  };
+};
+
+const rewriteBodyRes: RewriteRequestCase<IExecutionLogResult> = ({ data, total }: any) => ({
+  data: data.map((log: IExecutionLog) => getRenamedLog(log)),
+  total,
+});
+
+const getFilter = (filter: string[] | undefined) => {
+  if (!filter || !filter.length) {
+    return;
+  }
+  return filter.join(' OR ');
+};
+
+export type SortFields =
+  | 'timestamp'
+  | 'execution_duration'
+  | 'total_search_duration'
+  | 'es_search_duration'
+  | 'schedule_delay'
+  | 'num_triggered_actions';
+
+export type SortOrder = 'asc' | 'desc';
+
+export type SortField = Record<
+  SortFields,
+  {
+    order: SortOrder;
+  }
+>;
+
+export interface LoadExecutionLogAggregationsProps {
+  id: string;
+  dateStart: string;
+  dateEnd?: string;
+  filter?: string[];
+  perPage?: number;
+  page?: number;
+  sort?: SortField[];
+}
+
+export const loadExecutionLogAggregations = async ({
+  id,
+  http,
+  dateStart,
+  dateEnd,
+  filter,
+  perPage = 10,
+  page = 0,
+  sort = [],
+}: LoadExecutionLogAggregationsProps & { http: HttpSetup }) => {
+  const sortField: any[] = sort;
+
+  const result = await http.get<AsApiContract<IExecutionLogResult>>(
+    `${INTERNAL_BASE_ALERTING_API_PATH}/rule/${id}/_execution_log`,
+    {
+      query: {
+        date_start: dateStart,
+        date_end: dateEnd,
+        filter: getFilter(filter),
+        per_page: perPage,
+        page: page + 1,
```

Comment:
Probably worth adding a comment explaining why we have to add 1 - honestly, I don't know why [the API works this way](https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/lib/get_execution_log_aggregation.ts#L117) when it just subtracts `1` from the `page` variable later. I'd add something here to remind us about this.